python-threatexchange/threatexchange/hashing/pdq_faiss_matcher.py

```diff
@@ -101,6 +101,52 @@ def search(
             for i in range(len(query_vectors))
         ]
 
+    def search_with_distance(
+        self,
+        queries: t.Sequence[PDQ_HASH_TYPE],
+        threshhold: int,
+        return_as_ids: bool = False,
```

Comment:
(Nit): Hmm, this makes the return of this impossible to type, and as a result is slightly smelly to me. If you wanted to fix it, would suggest having two methods:

1. Return with ids
2. Return without ids (impl is to call with ids and then strip them)

Reply:
I agree, this is not part of these changes however I am trying to add what I need in  `threatexchange/hashing/pdq_faiss_matcher.py` without rewritting too much, however I can make this change for `search_with_distance_in_result` methods